kpatch-build/kpatch-build

```diff
@@ -807,14 +813,24 @@ grep -q "CONFIG_DEBUG_INFO_SPLIT=y" "$CONFIGFILE" && die "kernel option 'CONFIG_
 grep -q "CONFIG_GCC_PLUGIN_LATENT_ENTROPY=y" "$CONFIGFILE" && die "kernel option 'CONFIG_GCC_PLUGIN_LATENT_ENTROPY' not supported"
 grep -q "CONFIG_GCC_PLUGIN_RANDSTRUCT=y" "$CONFIGFILE" && die "kernel option 'CONFIG_GCC_PLUGIN_RANDSTRUCT' not supported"
 
+check_pahole_supports_btf
+
 # CONFIG_DEBUG_INFO_BTF invokes pahole, for which some versions don't
 # support extended ELF sections.  Disable the BTF typeinfo generation in
 # link-vmlinux.sh since kpatch doesn't care about that anyway.
-if grep -q "CONFIG_DEBUG_INFO_BTF=y" "$CONFIGFILE" ; then
+if [[ $(grep -q "CONFIG_DEBUG_INFO_BTF=y" "$CONFIGFILE") ]] && [[ "$PAHOLE_SUPPORTS_BTF" -eq 0 ]]; then
 	cp -f "$SRCDIR/scripts/link-vmlinux.sh" "$TEMPDIR/link-vmlinux.sh" || die
 	sed -i 's/CONFIG_DEBUG_INFO_BTF/DISABLED_FOR_KPATCH_BUILD/g' "$SRCDIR"/scripts/link-vmlinux.sh || die
 fi
 
+# CONFIG_DEBUG_INFO_BTF_MODULES invokes pahole for kernel modules, which
+# doesn't work with kpatch-cc.  Disable this command by replacing the line
+# with ": \".
+if [[ $(grep -q "CONFIG_DEBUG_INFO_BTF_MODULES=y" "$CONFIGFILE") ]] && [[ "$PAHOLE_SUPPORTS_BTF" -eq 0]]; then
+	cp -f "$SRCDIR/scripts/Makefile.modfinal" "$TEMPDIR/Makefile.modfinal" || die
+	sed -i 's/.*LLVM_OBJCOPY.*/\t\t:\t\\/g' "$SRCDIR"/scripts/Makefile.modfinal || die
```

Comment:
Instead of trying to fix this command which can change why not do the same thing as `CONFIG_DEBUG_INFO_BTF` code does and just disable everything?

`sed -i 's/CONFIG_DEBUG_INFO_BTF_MODULES/DISABLED_FOR_KPATCH_BUILD/g' "$SRCDIR"/scripts/Makefile.modfinal || die`

Reply:
Fixed this part and dropped the second commit. Thanks!